### ISSUE TITLE
gpg: keyserver receive failed: No data

### ISSUE THREAD

**Issue by DiyVE** [LionSec/katoolin#178]
Hello everybody,

When I try to add kali linux repositories, I have a message like this : 
`Executing: /tmp/apt-key-gpghome.mxXJwBhq3s/gpg.1.sh --keyserver pgp.mit.edu --recv-keys ED444FF07D8D0BF6
gpg: keyserver receive failed: No data`

Can you help me, i really want to have kali linux tool into my Ubuntu environment ;)

**Comment by developpeur62** [comment]
use wget -q -O - archive.kali.org/archive-key.asc | apt-key add
and sudo apt-get update

**Comment by DiyVE** [comment]
Thank you Anthony, but I have another problem now, and i think it have some relation with the failed installation of katoolin. Since few day I cannot connect my laptop on my network, even if i try to plug-in with the RJ-45 cable. When I try to connect the the wifi, my compter can see my wifi network but when i try to connect to it, it just show "connecting" and few seconds later an error message !!

**Comment by sagoode** [comment]
I'm having the same issue even **after** the wget command.
Executing: /tmp/apt-key-gpghome.z0xzfs32B1/gpg.1.sh --keyserver pgp.mit.edu --recv-keys ED444FF07D8D0BF6
gpg: keyserver receive failed: No data

**Comment by developpeur62** [comment]
ok, you have solve this problem ?

**Comment by DiyVE** [comment]
Not really, i cannot test your command line because i cannot access to internet. But yes It's another problem.

**Comment by felipelinuxbr** [comment]
apt-key-gpghome

Em Sáb, 9 de fev de 2019 16:11, DiyVE <notifications@github.com escreveu:

**Comment by sakuyaslove** [comment]
Ubuntu 18.04.2 LTS Fresh install updated apt-get installed and updated git and python 2.7 and pip

I get the same error.

@developpeur62 I used "use wget -q -O - archive.kali.org/archive-key.asc | apt-key add"
Received an OK message

Still get the following issue getting the repo:
Executing: /tmp/apt-key-gpghome.nQfL5ZOGok/gpg.1.sh --keyserver pgp.mit.edu --recv-keys ED444FF07D8D0BF6
gpg: keyserver receive failed: No data

**Comment by sakuyaslove** [comment]
I just tried it again and got this:

Executing: /tmp/apt-key-gpghome.VCcVuYIqHt/gpg.1.sh --keyserver pgp.mit.edu --recv-keys ED444FF07D8D0BF6
gpg: key ED444FF07D8D0BF6: 2 duplicate signatures removed
gpg: key ED444FF07D8D0BF6: 161 signatures not checked due to missing keys
gpg: key ED444FF07D8D0BF6: "Kali Linux Repository <devel@kali.org>" 164 new signatures
gpg: Total number processed: 1
gpg:         new signatures: 164

Maybe its just the pgp.mit.edu system?

**Comment by sakuyaslove** [comment]
something was wrong with installing the categories so I deleted the repo and tried again and now I am getting the same error about No Data. Wondering if its the mit system

**Comment by developpeur62** [comment]
me, i have installed this project with:
- wget -q -O - archive.kali.org/archive-key.asc | sudo apt-key add
- sudo apt-get update
- sudo apt-get install uniscan (
example)

**Comment by developpeur62** [comment]
you use ubuntu ?@sakuyaslove

**Comment by sakuyaslove** [comment]
@developpeur62 Yes I use Ubuntu, I did another fresh install of Ubuntu and was waiting to see what gets said on here. Do I need to try your wget method listed 4 days ago?

**Comment by developpeur62** [comment]
yes

**Comment by fabiozzug** [comment]
> me, i have installed this project with:
> 
> * wget -q -O - archive.kali.org/archive-key.asc | sudo apt-key add
> * sudo apt-get update
> * sudo apt-get install uniscan (
>   example)

i got another error  gpg: no valid OpenPGP data found

**Comment by OscarGonzalez97** [comment]
> @developpeur62 Yes I use Ubuntu, I did another fresh install of Ubuntu and was waiting to see what gets said on here. Do I need to try your wget method listed 4 days ago?

It works for you? I am with the same problem, and I don't know why.
I'm in Ubuntu 18.04.2 LTS

**Comment by fabiozzug** [comment]
> > @developpeur62 Yes I use Ubuntu, I did another fresh install of Ubuntu and was waiting to see what gets said on here. Do I need to try your wget method listed 4 days ago?
> 
> It works for you? I am with the same problem, and I don't know why.
> I'm in Ubuntu 18.04.2 LTS
https://mko-x.github.io/katoolin/ hay mate use this tool its katoolin v2 i fix the problem

**Comment by DGinUK** [comment]
use this

https://github.com/Rajkumrdusad/Tool-X

**Comment by 1337-ishaan** [comment]
` sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys ED444FF07D8D0BF6 `

Try this out, works for me

**Comment by OpAmlee** [comment]
ishaan that key worked for me. thank you
